data_catalog_backend/services/category_service.py

```diff
@@ -43,3 +52,42 @@ def create_category(self, category: CategoryRequest, user: User) -> Category:
             self.session.rollback()
             raise e
         return category
+
+    def delete_category(self, category_id: uuid.UUID, current_user: User) -> Category:
```

Comment:
I would prefer that this function doesn't return anything. It should raise Errors when the necessary conditions for deleting a resource isn't met.